Src/Base/AMReX_GpuAtomic.H

```diff
@@ -526,6 +526,136 @@ namespace detail {
         ))
 #endif
     }
+
+////////////////////////////////////////////////////////////////////////
+//  Multiply
+////////////////////////////////////////////////////////////////////////
+
+#ifdef AMREX_USE_GPU
+
+    template<class T>
+    AMREX_GPU_DEVICE AMREX_FORCE_INLINE
+    T Multiply_device (T* const /*prod*/, T const /*value*/) noexcept
+    {
+        amrex::Abort("Gpu::Atomic::Multiply is not implemented for this data type.");
```

Comment:
It might be better to remove this function template. It's better to have a compile time error than a run time error.

Reply:
If we want, we could also implement this in a different way to support user defined types in case they have overloaded operators. Something like
```
template <typename T, std::enable_if_t<sizeof(T) == sizeof(int), int> = 0>
AMREX_GPU_DEVICE AMREX_FORCE_INLINE
T Multiply_device (T* const prod, T const value) noexcept
{
    return detail::atomic_op<T, int>(prod,value,amrex::Multiplies<T>());
}

template <typename T, std::enable_if_t<sizeof(T) == sizeof(unsigned long long), int> = 0>
AMREX_GPU_DEVICE AMREX_FORCE_INLINE
T Multiply_device (T* const prod, T const value) noexcept
{   
    return detail::atomic_op<T, unsigned long long>(prod,value,amrex::Multiplies<T>());
}
```